source/game/field/ObjectDirector.cc

```diff
@@ -0,0 +1,55 @@
+#include "ObjectDirector.hh"
+
+#include "game/system/CourseMap.hh"
+#include "game/system/map/MapdataGeoObj.hh"
+
+namespace Field {
+
+ObjectDirector *ObjectDirector::CreateInstance() {
+    assert(!s_instance);
+    s_instance = new ObjectDirector;
+
+    s_instance->createObjects();
+
+    return s_instance;
+}
+
+void ObjectDirector::DestroyInstance() {
+    assert(s_instance);
+    delete s_instance;
+    s_instance = nullptr;
+}
+
+ObjectDirector *ObjectDirector::Instance() {
+    return s_instance;
+}
+
+ObjectDirector::ObjectDirector() = default;
+
+ObjectDirector::~ObjectDirector() = default;
+
+void ObjectDirector::createObjects() {
+    const auto *courseMap = System::CourseMap::Instance();
+    size_t objectCount = courseMap->getGeoObjCount();
+    m_objects.reserve(objectCount);
+
+    for (size_t i = 0; i < objectCount; ++i) {
+        const auto *pObj = courseMap->getGeoObj(i);
+        assert(pObj);
+
+        // Assume one player - if the presence flag isn't set, don't construct it
+        if (!(pObj->presenceFlag() & 1)) {
+            continue;
+        }
+
+        m_objects.push_back(createObject(*pObj));
+    }
+}
+
+Object *ObjectDirector::createObject(const System::MapdataGeoObj &params) {
+    return new Object(params);
```

Comment:
This is just a temporary function body right? This is basically our factory function, where we return the base `Object *` but really we'll be loading whatever object corresponds with the `GeoObj`'s id?

Reply:
Yes.